log4j-core/src/main/java/org/apache/logging/log4j/core/async/DisruptorUtil.java

```diff
@@ -82,35 +84,31 @@ static int calculateRingBufferSize(final String propertyName) {
     }
 
     static ExceptionHandler<RingBufferLogEvent> getAsyncLoggerExceptionHandler() {
-        final String cls = PropertiesUtil.getProperties().getStringProperty("AsyncLogger.ExceptionHandler");
-        if (cls == null) {
-            return new AsyncLoggerDefaultExceptionHandler();
-        }
+        ExceptionHandler<RingBufferLogEvent> handler = null;
         try {
-            @SuppressWarnings("unchecked")
-            final Class<? extends ExceptionHandler<RingBufferLogEvent>> klass =
-                (Class<? extends ExceptionHandler<RingBufferLogEvent>>) Loader.loadClass(cls);
-            return klass.newInstance();
-        } catch (final Exception ignored) {
-            LOGGER.debug("Invalid AsyncLogger.ExceptionHandler value: error creating {}: ", cls, ignored);
-            return new AsyncLoggerDefaultExceptionHandler();
+            handler =
```

Comment:
You don't need a `handler` variable'; you can simply place a `return` statement here and at the `catch` return the default.

Reply:
[`LoaderUtil.newCheckedInstanceOfProperty`](https://github.com/apache/logging-log4j2/blob/ce3a6de7a53a879764dadf01aae35298a039e6be/log4j-api/src/main/java/org/apache/logging/log4j/util/LoaderUtil.java#L330) can return `null` if the property is not defined. Since instantiation can also fail, leaving the variable at its default value of `null`, I do a single null-check for both cases.

Reply:
I added a version of `newCheckedInstanceOfProperty` that takes a `Supplier<T>` as additional parameter to provide a default value.

However in `3.x` we should probably consider returning `Optional<T>`: this way we can always call `Optional#orElseGet(Supplier<T>)` without the need of two versions of each method.